db/migrate/20181024163850_add_source_type.rb

```diff
@@ -0,0 +1,15 @@
+class AddSourceType < ActiveRecord::Migration[5.1]
+  def change
+    create_table :source_types, :id => :bigserial do |t|
```

Comment:
Should this be more generic?  Maybe json_schemas?  I'm thinking we may reuse this on Endpoints and Authentications, right?

Reply:
That would kind of pollute the table if json_schemas.name could include "openshift" and "ssh" without some other type as well.  Maybe we need an authentication_types also?

Reply:
Agree with @agrare here, even if we don't need separate tables for what we're storing I think this is the better approach. I would rather have a few tables with the same kind of data if it avoids the kind of confusion we would need to deal with to solve a few special cases.

Reply:
> Is a single json schema is enough to handle everything we need for a provider screen? Thinking possible endpoints, authentications types, etc...

Reread this comment ... is the json schema in this table also going to include the spec to create the source type's authentication and endpoint? Or is that going to be separate?

If it's all rolled into one schema then I think what we have here is what @bdunne is asking for right?

Reply:
> Reread this comment ... is the json schema in this table also going to include the spec to create the source type's authentication and endpoint? Or is that going to be separate?

Yeah that is what I'm wondering, I'm not sure how the UI will use this.  If all in one schema works then great, otherwise we'll have to specify what endpoint/auth types a source_type needs here, then we'll have to go to an endpoint_type.schema.

@martinpovolny do you know if a single json spec for a provider create screen will work?

Reply:
I was expecting separate json schemas for sources and endpoints (and sources referring to endpoints).

Reply:
Okay we can handle endpoint type json schema in a follow-up pr